app/main.py

```diff
@@ -1,29 +1,72 @@
+import json
 import logging
 import sys
 
+from fastapi.exceptions import HTTPException, RequestValidationError
 from fastapi.logger import logger
 from fastapi.openapi.utils import get_openapi
+from fastapi.requests import Request
+from fastapi.responses import JSONResponse
 from fastapi.staticfiles import StaticFiles
 from starlette.middleware.base import BaseHTTPMiddleware
 
 from app.routes.tasks import tasks
 
 from .application import app
+from .errors import ClientError, ServerError
 from .middleware import redirect_latest, set_db_mode
 from .routes import security
 from .routes.features import features
 from .routes.geostore import geostore
 from .routes.meta import assets, datasets, versions
 from .routes.sql import queries
 
-###############
+################
 # LOGGING
 ################
 
 gunicorn_logger = logging.getLogger("gunicorn.error")
 logger.handlers = gunicorn_logger.handlers
 sys.path.extend(["./"])
 
+
+################
+# ERRORS
+################
+
+
+@app.exception_handler(ClientError)
+async def client_error_handler(request: Request, exc: ClientError):
+    return JSONResponse(
+        status_code=exc.status_code, content={"status": "failed", "data": exc.detail}
+    )
+
+
+@app.exception_handler(ServerError)
+async def server_error_handler(request: Request, exc: ServerError):
+    return JSONResponse(
+        status_code=exc.status_code, content={"status": "error", "message": exc.detail}
+    )
+
+
+@app.exception_handler(HTTPException)
+async def httpexception_error_handler(request: Request, exc: HTTPException):
+    if exc.status_code < 500:
+        status = "failed"
+    else:
+        status = "error"
+    return JSONResponse(
+        status_code=exc.status_code, content={"status": status, "message": exc.detail}
+    )
+
+
+@app.exception_handler(RequestValidationError)
+async def rve_error_handler(request: Request, exc: RequestValidationError):
+    return JSONResponse(
+        status_code=400, content={"status": "failed", "message": json.loads(exc.json())}
```

Comment:
I believe response code for this is currently `422`, right?

Reply:
Fixed.